### PR TITLE
Holden/develop

### PR DESCRIPTION
**Pull request by nickclaw**

### CODE REVIEW

File: src/azuqua.js [hunk 1]
```diff
@@ -0,0 +1,847 @@
...
+    } catch(e) { 
+      let errProxy = new Error('Error mapping query string to string values');
+      return errProxy;
+    }
+
+
+    let timestamp = new Date().toISOString();
+
+    let meta = [method.toLowerCase(), path, timestamp].join(':') + pathQueryString;
+    let hash = crypto.createHmac('sha256', accessSecret).update(new Buffer(meta, 'utf-8')).digest('hex');
```
Comment (nickclaw) [inline src/azuqua.js]: `new Buffer()` is deprecated 

https://nodejs.org/api/buffer.html#buffer_new_buffer_array

File: src/azuqua.js [hunk 2]
```diff
@@ -0,0 +1,847 @@
...
+class Flo {
+  /**
+   * Creates a Flo instance with given Flo data
+   * @constructor
+   * @param {Object} floObject - Object containing the data to build the flo with
+   */
+  constructor(floObject) {
+    _.extend(this, floObject);
+  }
+}
```
Comment (nickclaw) [inline src/azuqua.js]: Not sure what the point of this class is..

File: src/azuqua.js [hunk 3]
```diff
@@ -0,0 +1,847 @@
...
+        let responseJson = JSON.parse(text);
+        return Promise.resolve(responseJson);
+      } catch (e) {
+        return Promise.reject({ 
+          type : 'ResponseParsingError',
+          message : `Error trying to parse the following server response as JSON: ${text}`
+        });
+      }
+    }) 
+}
```
Comment (stegmanh) [inline src/azuqua.js]: Added b/c I wanted a way to add some additional information if the parsing fails. Suppose I could catch that error down the line though.

File: src/azuqua.js [hunk 4]
```diff
@@ -0,0 +1,847 @@
...
+    this.protocol = 'https';
+    this.httpOptions = {
+      host: 'api.azuqua.com',
+      port: 443
+    };
+
+    // Cached Flos
+    this._flos = null;
+
+    if (arguments.length === 0 || arguments.length === 1) {
```
Comment (stegmanh) [inline src/azuqua.js]: Yeah good point on this, I'll think about it.

**Comment by stegmanh** [comment]
Thanks for code review!

**Pull request closed without merging**